src/main/java/com/uditagarwal/cache/Cache.java

```diff
@@ -25,6 +25,7 @@ public void put(Key key, Value value) {
                 throw new RuntimeException("Unexpected State. Storage full and no key to evict.");
             }
             this.storage.remove(keyToRemove);
+            System.out.println("Removing Least Recently used item..." + keyToRemove);
```

Comment:
Eviction might not be LRU based here.  Lets use generic thing like "Create space by evicting one item."